src/main/java/org/example/controllers/JobRoleController.java

```diff
@@ -31,4 +33,16 @@ public Response getListOpenJobRoles() {
             return Response.serverError().build();
         }
     }
+
+    @GET
+    @Path("/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getJobRoleById(@PathParam("id") final int id) {
+        try {
+            return Response.status(HttpStatus.OK_200)
+                    .entity(jobRoleService.getJobRoleById(id)).build();
+        } catch (JobRoleDaoException e) {
+            return Response.serverError().build();
+        }
+    }
```

Comment:
and then another question here, how do we want to handle the case when ids are not in the db? From an API perspective, 404 makes sense

Reply:
Also what if I send in an id that isn't an id?

Reply:
changed the catch to return a 404, is that enough error handling or does this require another catch?